src/uu/env/src/env.rs

```diff
@@ -87,6 +92,55 @@ fn parse_program_opt<'a>(opts: &mut Options<'a>, opt: &'a OsStr) -> UResult<()>
     }
 }
 
+fn parse_signal_value(signal_name: &str) -> UResult<usize> {
+    let signal_name_upcase = signal_name.to_uppercase();
+    let optional_signal_value = signal_by_name_or_value(&signal_name_upcase);
+    let error = USimpleError::new(125, format!("{}: invalid signal", signal_name.quote()));
+    match optional_signal_value {
+        Some(sig_val) => {
+            if sig_val == 0 {
+                Err(error)
+            } else {
+                Ok(sig_val)
+            }
+        }
+        None => Err(error),
+    }
+}
+
+fn parse_signal_opt<'a>(opts: &mut Options<'a>, opt: &'a OsStr) -> UResult<()> {
+    if opt.is_empty() {
+        return Ok(());
+    }
+    let signals: Vec<&'a OsStr> = opt
+        .as_bytes()
+        .split(|&b| b == b',')
+        .map(OsStr::from_bytes)
+        .collect();
+
+    let mut sig_vec = Vec::with_capacity(signals.len());
+    signals.into_iter().for_each(|sig| {
+        if !(sig_vec.contains(&sig) || sig.is_empty()) {
+            sig_vec.push(sig);
```

Comment:
Please move the call to `parse_signal_value` into this loop, and apply deduplication directly on `opts.ignore_signal`.

This is because there are thousands(\*) of ways to spell sigterm,SiGtErM,tERM,15,siGTerM, etc., and then the `sig_vec.contains`-deduplication would be entirely ineffective.

(\*) 1 + 2^4 + 2^7 = 144 for SIGTERM, 577 for SIGVTALRM, again 577 for SIGSTKFLT, …